mr/src/main/java/org/opensearch/hadoop/rest/commonshttp/CommonsHttpTransport.java

```diff
@@ -635,13 +667,21 @@ public Response execute(Request request) throws IOException {
         }
 
         ByteSequence ba = request.body();
+        byte[] bodyBytes = null;
         if (ba != null && ba.length() > 0) {
             if (!(http instanceof EntityEnclosingMethod)) {
-                throw new IllegalStateException(String.format("Method %s cannot contain body - implementation bug", request.method().name()));
+                throw new IllegalStateException(
+                        String.format("Method %s cannot contain body - implementation bug", request.method().name()));
             }
             EntityEnclosingMethod entityMethod = (EntityEnclosingMethod) http;
             entityMethod.setRequestEntity(new BytesArrayRequestEntity(ba));
             entityMethod.setContentChunked(false);
+
+            try (ByteArrayOutputStream outStream = new ByteArrayOutputStream()) {
+                ba.writeTo(outStream);
+                bodyBytes = outStream.toByteArray();
+                outStream.close();
+            }
```

Comment:
1. We should avoid copy the input ByteSequence.
2. https://github.com/opensearch-project/opensearch-hadoop/blob/main/mr/src/main/java/org/opensearch/hadoop/util/BytesArrayPool.java#L54 does not implement writeTo, is there any issue?

My suggestion is add InputStream toInputStream() interface to ByteSequence class.